### PR TITLE
DIG-1778: Tax Calculation in Spree Adjustments

### PR DESCRIPTION
**Pull request by vicmaster**
- Update gem to support positive adjustments and tax recalculation after that.
- Add this positive adjustment as a charge

### CODE REVIEW

File: app/models/spree/order_decorator.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ def commit_avatax_invoice(doc_type)
         line_count = 0
         discount = 0
         discount = calculate_order_discounts
+        charge = 0
```
Comment (tomfbiz) [inline app/models/spree/order_decorator.rb]: what is this line for? it seems to be overwritten in the next line.

File: app/models/spree/order_decorator.rb [hunk 2]
```diff
@@ -45,8 +48,22 @@ def commit_avatax_invoice(doc_type)
...
+
         invoice_lines << invoice_line
 
+        charge_line = Avalara::Request::Line.new(
+            :line_no => (line_count + 1).to_s,
+            :destination_code => '1',
+            :origin_code => '1',
+            :qty => 1,
+            :amount => charge.to_s,
+            :tax_code => 'FR',
```
Comment (tomfbiz) [inline app/models/spree/order_decorator.rb]: what does this tax code mean? maybe you should use a constant, or at least add a comment.

File: app/models/spree/order_decorator.rb [hunk 3]
```diff
@@ -45,8 +48,22 @@ def commit_avatax_invoice(doc_type)
...
+            :discounted => true,
+            :item_code => 'CHARGE'
+        )
+
+        invoice_lines << charge_line if charge > 0
+
```
Comment (tomfbiz) [inline app/models/spree/order_decorator.rb:68]: Avalara calls take time and cost money. I don't know if _Avalara::Request::Line.new_  makes an API call or not.
Should we be calling this all the time or just if charge > 0?

I would have something like 

```
if charge 
    line_count += 1
     charge_line = Avalara::Request::Line.new(
         :line_no => (line_count ).to_s,
         :destination_code => '1',
            .
            .
            .
       )

     invoice_lines << charge_line
end
```

 I think that would be clearer, and maybe faster. Also, if we ever add yet another line, the line_count would be correct.

File: app/models/spree/order_decorator.rb [hunk 4]
```diff
@@ -19,6 +19,8 @@ def commit_avatax_invoice(doc_type)
         line_count = 0
         discount = 0
         discount = calculate_order_discounts
+        charge = 0
+        charge = calculate_order_charge
```
Comment (tomfbiz) [inline app/models/spree/order_decorator.rb]: _charge_ does not seem like a very descriptive variable name. Everything is a charge that is not a credit.
Is this really something more like _adjustment_charges_?

**Comment by vicmaster** [comment]
@tomfbiz Could you review it again?, please

**Comment by tomfbiz** [comment]
@vicmaster you addressed all of my comments.
If Thomas can confirm by testing that this fixes the issue then this is good to go, in my opinion.

**Comment by vicmaster** [comment]
@tomfbiz Thomas said "is good to go".

**Comment by tomfbiz** [comment]
LGTM

**Pull request merged**